packages/client/shared/gqlIds/IntegrationRepoId.ts

```diff
@@ -11,16 +12,22 @@ type JiraRepoIntegration = {
   service: 'jira'
 }
 
+export type RepoIntegration = {
+  id: string
+  providerId: string
+  service: Exclude<IntegrationProviderServiceEnum, 'jira' | 'github'>
+}
+
 const IntegrationRepoId = {
-  join: (integration: GitHubRepoIntegration | JiraRepoIntegration) => {
+  join: (integration: GitHubRepoIntegration | JiraRepoIntegration | RepoIntegration) => {
     const {service} = integration
     switch (service) {
       case 'github':
         return integration.nameWithOwner
       case 'jira':
         return JiraProjectId.join(integration.cloudId, integration.projectKey)
       default:
-        return ''
+        return `${integration.service}:${integration.providerId}:${integration.id}`
```

Comment:
I'm uncertain about this bit. I like that with the service prefixed, we always can split it without having to pass the `service` separately, but it's redundant since it's already stored in the provider.